src/TwigComponent/src/ComponentRenderer.php

```diff
@@ -105,6 +105,7 @@ private function preRender(MountedComponent $mounted, array $context = []): PreR
 
             // expose public properties and properties marked with ExposeInTemplate attribute
             iterator_to_array($this->exposedVariables($component, $metadata->isPublicPropsExposed())),
+            $component instanceof AnonymousComponent ? $component->getProps() : []
```

Comment:
WDYT about somehow reusing `#[ExposeInTemplate]` to do this, instead of this special case? We would need some new option on that attribute, which would mean "I am returning an associative array of variable names" - I'm not sure about a good name:

```php
#[ExposeInTemplate(destructure: true)
public function getProps(): array
{
    return $this->props;
}
```

Reply:
I made something, tell me what you think about it